mtt/src/mtt.rs

```diff
@@ -74,6 +75,24 @@ mod mtt_const {
     pub const L2_TYPE_MASK: u64 = 0x000c_0000_0000;
     // Bits 33:0
     pub const L2_INFO_MASK: u64 = 0x0003_ffff_ffff;
+}
+
+// This is to help facilitate testing. The issue is that when testing, we use the PFN
+// of the VA for the L1-page, and since it tends to be very large (unlike the actual PA),
+// it might set bits 35:34 when converting to a L1 page (thereby converting the L2 entry
+// into a different type). We avoid the issue by shifting everything left by 4.
```

Comment:
this still feels extremely brittle to me :/

another option -- which i don't necessarily want to gate submitting this PR on, but i think would be preferable to the one here -- is to hunt for a place in the low 46-bits of VA space using `mmap(MAP_FIXED_NOREPLACE)`.

Reply:
Good suggestion...changed.